votes/0606-pr-issue-association.md

```diff
@@ -0,0 +1,52 @@
+# A Vote for PR Issue Association
+
+## Proposal
+
+### Motivation
+
+To measure software evolution, we focus on features implemented, bugs fixed, and general enhancement. Issues document these items, while a pull request focuses on the change set and whether we implement it correctly.
+
+Normally, an issue documents a work item, and a pull request resolves it (despite cherry-picks). However, the relation between pull requests and issues is complicated in TiDB projects.
+
+- 1 issue with 0 PR: question, bug vanished without a reason, etc. 
+- 1 issue with 1 PR: normal case. 
+- 1 issue with n PRs: resolve a bug multiple times([example](https://github.com/pingcap/tidb/issues/24679)), or resolve a complex issue in steps([example](https://github.com/pingcap/tidb/issues/27652)). 
+- n issues with 1 PR: trade off resolving things as a whole (due to code coupling, or we can resolve multiple bugs at the common parent level)([example](https://github.com/pingcap/tidb/pull/27697)), duplicated issues. 
+- 0 issue with 1 PR: coding without a ticket([example](https://github.com/pingcap/tidb/pull/23022)). Even if the description is detailed or even if a design document is present, pull requests focus on the change set. It is about the implementation, not the question; if it is, we always mix discussion because the change set has a primary position. 
+
+This proposal tries to resolve the last case (0 issue with 1 PR), that a pull request must associate with issue(s). In this way,
+
+- Bring issues to a first-class development concept.
```

Comment:
For bug fixes, new feature or everything?
For example, I just want to fix a typo, and that PR still need a issue?

Reply:
At this time, yes. We are forcing the rule so there is no exception. I think such issues won't be too many. But in future, when most contributors are already acquainted with this rule and if such issues causes too much work for contributors, we could add a new rule(such as adding a `skip-issue` label like Python community) to skip this requirement.

Reply:
we can create some general purpose issues, such as “fix typo“, “code refactor”, “add test” to let PR refer them 🤔

Reply:
Flink has a 'hotfix' label to make certain simple fixes. However I think the rational behind this proposal is to make sure all release versions affected by an issue are properly fixed. So if such simple fixes affects multiple versions, it's better to have an issue to track.

Reply:
Make sense